infrastructure/repository/user_repository.go

```diff
@@ -0,0 +1,98 @@
+package repositories
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/cyneptic/letsgo-authentication/internal/core/entities"
+	"github.com/google/uuid"
+	
+	"gorm.io/gorm"
+)
+
+// check user already register to prevent to register twice
+func (p *Postgres) IsUserAlreadyRegisters(user entities.User) (int64, error) {
+	var count int64
+	res := p.db.Model(entities.User{}).Where("email = ?", user.Email).Count(&count)
+	if res.Error != nil {
+		if res.Error == gorm.ErrRecordNotFound {
+			return 0, nil // User not found, no error
+		}
+		print("auth respository 16", res.Error)
+		return 0, res.Error // Other error occurred
+	}
+	return count, nil
+}
```

Comment:
this can be handled by making the user's email or username a unique value in database.